libraries/AR_Motors/AP_MotorsUGV.cpp

```diff
@@ -292,8 +292,8 @@ bool AP_MotorsUGV::has_sail() const
 
 void AP_MotorsUGV::output(bool armed, float ground_speed, float dt)
 {
-    // soft-armed overrides passed in armed status
-    if (!hal.util->get_soft_armed()) {
+    // soft-armed and emergency stop overrides local armed status
+    if (!hal.util->get_soft_armed() || SRV_Channels::get_emergency_stop()) {
         armed = false;
```

Comment:
This change means were now tripping the disarmed actions. So we might start outputting 0 PWM depending on `SAFE_DISARM` param. It also means we don't bother doing speed scaling on steering. This would be bad for sailboats, I used to quite regularly sail round in E-stop on my boat with both a sail and motor.

Hence in this PR I did the same in a separate if statement. 

https://github.com/ArduPilot/ardupilot/pull/18612/files#diff-46e7a65ed2af1e80880e4279d1e05661388bed9bf44b2f725097b4e2ca5cdcaaR303

We should also be setting `_throttle_prev = 0.0` Otherwise the throttle will still be slewing from its previously high value when you un-E-stop.

Reply:
I'm not sure about the setting of _throttle_prev.  In my tests it slews properly.  I think maybe by setting _throttle_prev you'll make it so it doesn't throttle down slowly but instead jumps down.

I agree with you about not setting "armed = false" so your PR is better in that way at least.

I definitely don't think you should be sailing around with e-stop on a sailboat... sorry but that smacks of what the tradheli guys where they want to be able to fly while disarmed.. it's too big a rule to break so I don't think we should support that.. maybe it works but it shouldn't be a recommended or a new requirement for how it should function.